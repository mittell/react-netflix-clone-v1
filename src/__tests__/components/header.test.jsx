/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Header } from '../../components';

jest.mock('react-router-dom');

describe('<Header/>', () => {
	it('renders the basic <Header/> with a background', () => {
		const { container, getByText, getByTestId } = render(
			<Header>
				<Header.Frame>
					<Header.Logo src='/logo.png' alt='Aniflix' />
					<Header.TextLink active='true'>Hello I am a link!</Header.TextLink>
				</Header.Frame>
			</Header>
		);

		expect(getByText('Hello I am a link!')).toBeTruthy();
		expect(getByTestId('header-bg')).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});

	it('renders the basic <Header/> without a background', () => {
		const { container, getByText, queryByTestId } = render(
			<Header bg={false}>
				<Header.Frame>
					<Header.Logo src='/logo.png' alt='Aniflix' />
					<Header.ButtonLink>Sign In</Header.ButtonLink>
					<Header.TextLink active={false}>Hello I am a link!</Header.TextLink>
				</Header.Frame>
			</Header>
		);

		expect(getByText('Hello I am a link!')).toBeTruthy();
		expect(queryByTestId('header-bg')).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});

	it('renders the full <Header/> with a background', () => {
		const { container, getByText, getByTestId } = render(
			<Header src='evangelion-2' dontShowOnSmallViewPort>
				<Header.Frame>
					<Header.Group>
						<Header.Logo src='/images/logo.png' alt='Aniflix' />
						<Header.TextLink active={false} onClick={() => {}}>
							Series
						</Header.TextLink>
						<Header.TextLink active onClick={() => {}}>
							Films
						</Header.TextLink>
					</Header.Group>
					<Header.Group>
						<Header.Search searchTerm='Evangelion' setSearchTerm={() => {}} />
						<Header.Profile>
							<Header.Picture src='/images/users/1.png' />
							<Header.Dropdown>
								<Header.Group>
									<Header.Picture src='/images/users/1.png' />
									<Header.TextLink>User</Header.TextLink>
								</Header.Group>
								<Header.Group>
									<Header.TextLink onClick={() => {}}>Sign out</Header.TextLink>
								</Header.Group>
							</Header.Dropdown>
						</Header.Profile>
					</Header.Group>
				</Header.Frame>

				<Header.Feature>
					<Header.FeatureCallOut>
						Watch Neon Genesis Evangelion Now
					</Header.FeatureCallOut>
					<Header.Text>Fifteen years after...</Header.Text>
					<Header.PlayButton>Play</Header.PlayButton>
				</Header.Feature>
			</Header>
		);

		expect(getByTestId('search-input')).toBeTruthy();
		expect(getByTestId('search-input').value).toBe('Evangelion');
		fireEvent.change(getByTestId('search-input'), {
			target: { value: 'Test' },
		});
		fireEvent.click(getByTestId('search-click'));

		expect(getByText('Series')).toBeTruthy();
		expect(getByText('Films')).toBeTruthy();
		expect(getByText('User')).toBeTruthy();
		expect(getByText('Watch Neon Genesis Evangelion Now')).toBeTruthy();
		expect(getByText('Sign out')).toBeTruthy();
		expect(getByText('Fifteen years after...')).toBeTruthy();
		expect(getByText('Play')).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
