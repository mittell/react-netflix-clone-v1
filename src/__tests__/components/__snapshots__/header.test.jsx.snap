// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders the basic <Header/> with a background 1`] = `
<div
  class="header__Background-sc-nb0keb-0 dCYxwi"
  data-testid="header-bg"
>
  <div
    class="header__Container-sc-nb0keb-1 hWVUQq"
  >
    <p
      class="header__Link-sc-nb0keb-7 gbxYXX"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header/> renders the basic <Header/> without a background 1`] = `
<div
  class="header__Container-sc-nb0keb-1 hWVUQq"
>
  <p
    class="header__Link-sc-nb0keb-7 wUrsP"
  >
    Hello I am a link!
  </p>
</div>
`;

exports[`<Header/> renders the full <Header/> with a background 1`] = `
<div
  class="header__Background-sc-nb0keb-0 krNzDd"
  data-testid="header-bg"
  src="evangelion-2"
>
  <div
    class="header__Container-sc-nb0keb-1 hWVUQq"
  >
    <div
      class="header__Group-sc-nb0keb-8 HooCG"
    >
      <p
        class="header__Link-sc-nb0keb-7 wUrsP"
      >
        Series
      </p>
      <p
        class="header__Link-sc-nb0keb-7 wUrsP"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-sc-nb0keb-8 HooCG"
    >
      <div
        class="header__Search-sc-nb0keb-12 ejSkCn"
      >
        <button
          class="header__SearchIcon-sc-nb0keb-13 cgqljU"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-sc-nb0keb-14 fynrjX"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Evangelion"
        />
      </div>
      <div
        class="header__Profile-sc-nb0keb-11 gCXaHC"
      >
        <button
          class="header__Picture-sc-nb0keb-9 bTakSg"
          src="/images/users//images/users/1.png.png"
        />
        <div
          class="header__Dropdown-sc-nb0keb-10 bAJLSv"
        >
          <div
            class="header__Group-sc-nb0keb-8 HooCG"
          >
            <button
              class="header__Picture-sc-nb0keb-9 bTakSg"
              src="/images/users//images/users/1.png.png"
            />
            <p
              class="header__Link-sc-nb0keb-7 wUrsP"
            >
              User
            </p>
          </div>
          <div
            class="header__Group-sc-nb0keb-8 HooCG"
          >
            <p
              class="header__Link-sc-nb0keb-7 wUrsP"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-sc-nb0keb-1 header__Feature-sc-nb0keb-4 hWVUQq japCWc"
  >
    <h2
      class="header__FeatureCallOut-sc-nb0keb-6 dWozbI"
    >
      Watch Neon Genesis Evangelion Now
    </h2>
    <p
      class="header__Text-sc-nb0keb-5 hbCCCy"
    >
      Fifteen years after...
    </p>
    <button
      class="header__PlayButton-sc-nb0keb-15 lgrQwI"
    >
      Play
    </button>
  </div>
</div>
`;
